### PR TITLE
fix: Correctly detect development mode.

### PR DESCRIPTION
**Pull request by ShogunPanda**
This address an issue reported in #728.

### CODE REVIEW

File: packages/plugin-docusaurus-v3/src/index.ts [hunk 1]
```diff
@@ -58,7 +58,7 @@ export default function OramaPluginDocusaurus(
     },
 
     async allContentLoaded({ actions, allContent }) {
-      const isDevelopment = !options.cloud?.oramaCloudAPIKey
+      const isDevelopment = process.env.NODE_ENV === 'development' && !options.cloud?.oramaCloudAPIKey
```
Comment (micheleriva) [inline packages/plugin-docusaurus-v3/src/index.ts:61]: Would you turn that into a `||` for backward compatibility? Thanks!

Comment (ShogunPanda) [reply packages/plugin-docusaurus-v3/src/index.ts]: No problem! Done! What about now?

**Review by micheleriva — APPROVED** [review]

**Comment by micheleriva** [comment]
Sorry that error wasn't linked to your PR. Fixing separately

**Comment by ShogunPanda** [comment]
Cool, thanks sir! ^^

**Pull request merged**